File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] map = new int[26];
        
        for(int i=0;i<s.length();i++) map[s.charAt(i)-'a']++;
        for(int i=0;i<t.length();i++) map[t.charAt(i)-'a']--;
        
        for(int i=0;i<=map.length/2;i++) {
            if(map[i] != 0 || map[map.length-1-i] != 0) return false;
        }
        return true;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t): return False
        
        s_cnt = Counter(s)
        t_cnt = Counter(t)

        for c in s:
            if s_cnt[c] != t_cnt[c]:
                return False
        return True